nilearn/glm/contrasts.py

```diff
@@ -67,7 +67,7 @@ def compute_contrast(labels, regression_result, con_val, stat_type=None):
         Where q = number of :term:`contrast` vectors
         and p = number of regressors.
 
-    stat_type : {None, 't', 'F'}, optional
+    stat_type : {None, 't', 'F'}, optional, default=None
```

Comment:
```suggestion
    stat_type : {None, 't', 'F'}, default=None
```

Actually let's just not have 'optional' if we define the default.

Reply:
for this one and the other changes

Reply:
See example from our doc: https://nilearn.github.io/stable/development.html#coding-style

```python
def good(x, y=1, z=None):
    """Show how parameters are documented.

    Parameters
    ----------
    x : :obj:`int`
          X

    y : :obj:`int`, default=1
          Note that "default=1" is preferred to "Defaults to 1".

    z : :obj:`str`, default=None

    """


def bad(x, y=1, z=None):
    """Show how parameters should not be documented.

    Parameters
    ----------
    x :
          The type of X is not described

    y : :obj:`int`
          The default value of y is not described.

    z : :obj:`str`
          Defaults=None.
          The default value should be described after the type.
    """
```

Reply:
Also I have been noticing functions that have the same parameters but they're not on the list and they don't have the default values. Like in [nilearn/glm/second_level/second_level.py](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/second_level/second_level.py), the function [nilearn/glm/second_level/second_level.py::fit - line: 440](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/second_level/second_level.py#L440) also has the parameters 'confounds' and 'design_matrix'

Reply:
> Do you want me to remove "optional" wherever there is a default value defined in the module

If you come across some instances, feel free to remove the extra "optional" but do not make it your task to hunt and remove all instances.

Probably better to adapt our little script to flag those instances to be systematic about it.

Reply:
> Also I have been noticing functions that have the same parameters but they're not on the list and they don't have the default values. Like in [nilearn/glm/second_level/second_level.py](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/second_level/second_level.py), the function [nilearn/glm/second_level/second_level.py::fit - line: 440](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/second_level/second_level.py#L440) also has the parameters 'confounds' and 'design_matrix'

ha yeah I think our script only flags functions and does not flag methods.

so here too: if you come across some things that need changing, go ahead but don't make it your goal to catch them all, because from experience things can quickly turn into a bottom-less rabbit hole otherwise...